File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int first = 0;
        int last = nums.size()-1;
        int mid = 0;
        while(mid <= last){
            switch(nums[mid]){
                case 0:
                    swap(nums[first++], nums[mid++]);
                    break;
                case 1:
                    mid++;
                    break;
                case 2:
                    swap(nums[last--], nums[mid]);
                    break;
            }
        }
    }
};

File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int first = 0;
        int last = nums.length - 1;
        int current = 0;
        int temp;
        while(current <= last){
            switch(nums[current]){
                case 0:
                    temp = nums[first];
                    nums[first++] = nums[current];
                    nums[current++] = temp;
                    break;
                case 1:
                    current++;
                    break;
                case 2:
                    temp = nums[last];
                    nums[last--] = nums[current];
                    nums[current] = temp;
            }
        }
    }
}